control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -472,10 +472,13 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Reconcile hostedcontrolplane availability and Ready flag
 	{
+		kasExternalHostname := util.ServiceExternalDNSHostname(hostedControlPlane, hyperv1.APIServer)
+
 		infrastructureCondition := meta.FindStatusCondition(hostedControlPlane.Status.Conditions, string(hyperv1.InfrastructureReady))
 		kubeConfigAvailable := hostedControlPlane.Status.KubeConfig != nil
 		etcdCondition := meta.FindStatusCondition(hostedControlPlane.Status.Conditions, string(hyperv1.EtcdAvailable))
 		kubeAPIServerCondition := meta.FindStatusCondition(hostedControlPlane.Status.Conditions, string(hyperv1.KubeAPIServerAvailable))
+		kasExternalDNSHostnameReachable := kasExternalHostname == "" || util.IsUrlReachable(kasExternalHostname)
```

Comment:
Nit: this syntax is clever but weird to read.  Maybe this instead:
```
kasExternalDNSHostnameReachable := true
if kasExternalHostname != "" {
  kasExternalDNSHostnameReachable = util.IsUrlReachable(kasExternalHostname)
}
```
I know it is more code, but it is easier to understand imho